driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -64,7 +64,30 @@ public static Bson addFields(final Field<?>... fields) {
      * @since 3.4
      */
     public static Bson addFields(final List<Field<?>> fields) {
-        return new AddFieldsStage(fields);
+        return new FieldsStage("$addFields", fields);
+    }
+
+    /**
+     * Creates an $accumulator pipeline stage
+     *
+     * @param initField            a function used to initialize the state
+     * @param initArgsField        init function’s arguments (may be null)
+     * @param accumulateField      a function used to accumulate documents
+     * @param accumulateArgsField  additional accumulate function’s arguments (may be null). The first argument to the function is ‘state’.
+     * @param mergeField           a function used to merge two internal states, e.g. accumulated on different shards or threads. It returns
+     *                             the resulting state of the accumulator.
+     * @param finalizeField        a function used to finalize the state and return the result (may be null)
+     * @param lang                 a language specifier
+     * @return the $accumulator pipeline stage
+     * @mongodb.driver.manual reference/operator/aggregation/accumulator/ $accumulator
+     * @mongodb.server.release 4.4
+     * @since 4.1
+     */
+    public static Bson accumulator(final Field<?> initField, @Nullable final Field<?> initArgsField, final Field<?> accumulateField,
```

Comment:
Its conventional to have a method also that doesn't contain all the nullable fields. So lets add those methods.  Also as lang is only 'js' at the moment lets add a method that sets that automatically and a method that allows a user to override it in future.

Also, lets help the user with the API `Field` is quite generic and could easily let the user write the wrong code.  Where as init, accumulate, merge and finializer are all Strings and the args are all lists of strings?

So it would be more user friendly like so:

```
accumulator(String initFunction, String accumulateFunction, String mergeFunction)
accumulator(String initFunction, String accumulateFunction, String mergeFunction, String finializeFunction)
accumulator(String initFunction, String accumulateFunction, String mergeFunction, String finializeFunction, String lang)
...
accumulator(String initFunction, List<String> initArgs, 
                     String accumulateFunction,List<String> accumulateArgs,  
                     String mergeFunction, String finializeFunction, String lang)
```

Reply:
This is indeed more friendly for the user. I have added/updated the accumulator methods.